File: Erp.Server/Models/OrderMovementHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Erp.Server.Models
{
    public class OrderMovementHistory
    {
        [Key]
        [Display(Name = "Id")]
        public int omh_id { get; set; }

        [Display(Name = "Order No")]
        public int? omh_order_no { get; set; }

        [Display(Name = "Status")]
        public int? omh_status { get; set; }

       
        [Display(Name = "Status Name")]
        public string? omh_status_name { get; set; } = string.Empty;

        [Display(Name = "Created By")]
        public int? omh_cre_by { get; set; }

       
        [Display(Name = "Created By Name")]
        public string? omh_cre_by_name { get; set; } = string.Empty;

        [Display(Name = "Created On")]
        [DataType(DataType.DateTime)]
        public DateTime? omh_cre_date { get; set; } = DateTime.Now;
    }
}

File: erp_customer_mobile/lib/models/order_movement_history.dart
class OrderMovementHistory {
  final int omh_id;
  final int? omh_order_no;
  final int? omh_status;
  final String? omh_status_name;
  final int? omh_workflow_id;
  final int? omh_cre_by;
  final String? omh_cre_by_name;
  final DateTime? omh_cre_date;

  OrderMovementHistory({
    required this.omh_id,
    this.omh_order_no,
    this.omh_status,
    this.omh_status_name,
    this.omh_workflow_id,
    this.omh_cre_by,
    this.omh_cre_by_name,
    this.omh_cre_date,
  });

  factory OrderMovementHistory.fromJson(Map<String, dynamic> json) {
    return OrderMovementHistory(
      omh_id: json['omh_id'] ?? 0,
      omh_order_no: json['omh_order_no'],
      omh_status: json['omh_status'],
      omh_status_name: json['omh_status_name'],
      omh_workflow_id: json['omh_workflow_id'],
      omh_cre_by: json['omh_cre_by'],
      omh_cre_by_name: json['omh_cre_by_name'],
      omh_cre_date: json['omh_cre_date'] != null ? DateTime.parse(json['omh_cre_date']) : null,
    );
  }
}
